web3/utils/math.py

```diff
@@ -0,0 +1,11 @@
+import math
+
+
+def percentile(values, percentile):
+    if not values:
+        return None
+    index = len(values) * percentile / 100
+    if index % 2 == 0:
+        return (sorted(values)[index] + sorted(values)[index + 1]) / 2
+
+    return sorted(values)[math.ceil(index)]
```

Comment:
Sorting might be slow, as `values` gets bigger. It's probably worth sorting once and reusing the sorted list for performance, instead of sorting three times in the worst case.